spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java

```diff
@@ -191,9 +203,32 @@ else if (!CollectionUtils.isEmpty(apps)) {
 			registrations.addAll(appRegistry.importAll(force, bar));
 		}
 		Collections.sort(registrations);
+		prefetchMetadata(registrations);
 		return pagedResourcesAssembler.toResource(new PageImpl<>(registrations), assembler);
 	}
 
+	/**
+	 * Trigger early resolution of the metadata resource of registrations that have an explicit metadata artifact.
+	 * This assumes usage of {@link org.springframework.cloud.deployer.resource.support.DelegatingResourceLoader}.
+	 */
+	private void prefetchMetadata(List<AppRegistration> appRegistrations) {
+		forkJoinPool.execute(() -> {
+			appRegistrations.stream()
+				.filter(r -> r.getMetadataUri() != null)
```

Comment:
Does this mean during the prefetch, we don't download the artifact where the app itself contains the metadata resource? ie) only prefetch if there are explicit metadata resource available for the app registration?